observability/tracing.go

```diff
@@ -171,29 +173,74 @@ func getExporter(collectorURL string) (sdktrace.SpanExporter, error) {
 					)
 				} else {
 					log.Debugf("Creating OpenTelemetry collector with URL https://%s", collectorURL)
-
-					client = otlptracehttp.NewClient(otlptracehttp.WithEndpoint(collectorURL),
-						tracingOptions,
-					)
+					// That's mainly for testing
+					if tracingOpt.Otel.SkipVerify {
+						client = otlptracehttp.NewClient(otlptracehttp.WithEndpoint(collectorURL),
+							otlptracehttp.WithTLSClientConfig(&tls.Config{InsecureSkipVerify: true}),
+							tracingOptions,
+						)
+					} else {
+						client = otlptracehttp.NewClient(otlptracehttp.WithEndpoint(collectorURL),
+							tracingOptions,
+						)
+					}
 				}
 
 				ctx := context.Background()
-				exporter, err := otlptrace.New(ctx, client)
-				return exporter, err
+				httpExporter, err2 := otlptrace.New(ctx, client)
+				return httpExporter, err2
 			} else {
 				if tracingOpt.Otel.Protocol == GRPC {
 					log.Debugf("Creating OpenTelemetry grpc collector with URL %s", collectorURL)
 					ctx := context.Background()
-					ctx, cancel := context.WithTimeout(ctx, time.Second)
+					ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
 					defer cancel()
 
-					// TODO: Support TLS
-					exporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithInsecure(),
-						otlptracegrpc.WithEndpoint(collectorURL),
-						otlptracegrpc.WithDialOption(grpc.WithBlock()),
-					)
-
-					return exporter, err
+					var grpcExporter *otlptrace.Exporter
+
+					if tracingOpt.Otel.TlsEnabled {
+						var creds credentials.TransportCredentials
+						// That's mainly for testing
+						if tracingOpt.Otel.SkipVerify {
+							tlsConfig := &tls.Config{
```

Comment:
same here - do we want to TRACE log a message saying this is insecure?

Reply:
yes, I think is a good idea